From 11f6e8d02102d74356ad0d080c515909a5824f54 Mon Sep 17 00:00:00 2001
From: Nancy Chen <Nancy.Chen@freescale.com>
Date: Wed, 14 May 2008 09:34:45 -0500
Subject: [PATCH] ENGR00069620 SPI suspend issue

System can't go into low power mode due to error returned from SPI driver

Signed-off-by: Nancy Chen <Nancy.Chen@freescale.com>
---
 drivers/spi/mxc_spi.c |   20 --------------------
 1 files changed, 0 insertions(+), 20 deletions(-)

diff --git a/drivers/spi/mxc_spi.c b/drivers/spi/mxc_spi.c
index 570aac5..e0f5c09 100644
--- a/drivers/spi/mxc_spi.c
+++ b/drivers/spi/mxc_spi.c
@@ -1094,21 +1094,6 @@ static int mxc_spi_remove(struct platform_device *pdev)
 }
 
 #ifdef CONFIG_PM
-static int suspend_devices(struct device *dev, void *pm_message)
-{
-	pm_message_t *state = pm_message;
-
-	if (!dev->driver)
-		return 0;
-
-	if (dev->power.power_state.event != state->event) {
-		dev_warn(dev, "mismatch in pm state request\n");
-		return -1;
-	}
-
-	return 0;
-}
-
 static int spi_bitbang_suspend(struct spi_bitbang *bitbang)
 {
 	unsigned long flags;
@@ -1155,11 +1140,6 @@ static int mxc_spi_suspend(struct platform_device *pdev, pm_message_t state)
 	struct mxc_spi *master_drv_data = spi_master_get_devdata(master);
 	int ret = 0;
 
-	if (device_for_each_child(&pdev->dev, &state, suspend_devices) != 0) {
-		dev_warn(&pdev->dev, "suspend aborted\n");
-		return -EINVAL;
-	}
-
 	spi_bitbang_suspend(&master_drv_data->mxc_bitbang);
 	__raw_writel(MXC_CSPICTRL_DISABLE,
 		     master_drv_data->base + MXC_CSPICTRL);
-- 
1.5.4.4

